### ISSUE TITLE
`not()` fails with error

### ISSUE THREAD

**Issue by thasmo** [kenwheeler/cash#132]
Simplified example:
```js
var element = $(this);
var items = element.find('.item');
var item = items.first();
items.not(item).addClass('active');
```
Error:
```
SyntaxError: An invalid or illegal string was specified
```

Fails inside the function `matches` on the `return` line on column 0:
```js
function matches(el, selector) {
    var m = el && (el.matches || el.webkitMatchesSelector || el.mozMatchesSelector || el.msMatchesSelector || el.oMatchesSelector);
    return !!m && m.call(el, selector);
  }
```

**Comment by shshaw** [comment]
Ah, good catch. The issue is actually in `$.fn.not` where we need to account for cases where the argument provided is not a string, which is something we [fixed for `$.fn.is`](https://github.com/kenwheeler/cash/blob/3cb498ca7c5776b8cadcce5884f18e93a61931a9/src/traversal.js#L20-L36).

**Comment by shshaw** [comment]
Yikes, similar issues across a lot of these other traversal functions where they only accept a string selector and/or they're returning an Array instead of a Collection. Fixing these bugs as I run across them.

**Comment by shshaw** [comment]
I've pushed quite a few of these fixes to the `traversal-fixes` branch. Please give it a try and let me know if you encounter any oddities. I've fixed quite a few different items, so I'm not quite ready to push it to the main branch without further testing.

**Comment by thasmo** [comment]
Won't be able to test it on the project I've been working on for the next two weeks. Don't the tests cover the changes in the `traversal-fixes` branch properly?

**Comment by shshaw** [comment]
The test suite is not as robust as it should be :-)

The changes didn't break any existing tests (except one that was returning
wrong results, which I correct), but I didn't have time to build out test
cases for all the fixes.

Feel free to add some tests if you have the time & interest!

**Comment by shshaw** [comment]
Fixed in [1.3.5](https://github.com/kenwheeler/cash/releases/tag/1.3.5)
